File: Common/Settings/HardwareSettings.cs
/*
 * Copyright (c) 2022 Maximilian Bauer
 * Distributed under the GNU GPL v3. For full terms see the file "COPYING".
 */
#nullable enable

namespace GrowOne.Common.Settings
{
    public class HardwareSettings
    {
        public int MoistureSensorPin { get; set; } = 34;

        public int Dht22EchoPin { get; set; } = 12;

        public int Dht22TriggerPin { get; set; } = 14;

        public int Hcsr04EchoPin { get; set; } = 25;

        public int Hcsr04TriggerPin { get; set; } = 26;

        public int Hcsr04MinimumValueCm { get; set; } = 2;

        public int Hcsr04MaximumValueCm { get; set; } = 17;

        public int IrrigatorSwitchPin { get; set; } = 23;

        public int SensorSwitchPin { get; set; } = 22;

        public int BuzzerPin { get; set; } = 18;

        public string? Password { get; set; } = null;

        public HardwareSettings Clone()
        {
            return new HardwareSettings()
            {
                MoistureSensorPin = MoistureSensorPin,
                Dht22EchoPin = Dht22EchoPin,
                Dht22TriggerPin = Dht22TriggerPin,
                Hcsr04EchoPin = Hcsr04EchoPin,
                Hcsr04TriggerPin = Hcsr04TriggerPin,
                Hcsr04MinimumValueCm = Hcsr04MinimumValueCm,
                Hcsr04MaximumValueCm = Hcsr04MaximumValueCm,
                IrrigatorSwitchPin = IrrigatorSwitchPin,
                SensorSwitchPin = SensorSwitchPin,
                BuzzerPin = BuzzerPin,
                Password = Password
            };
        }
    }
}

File: GrowOne/Resources/WebResources/common/HardwareSettings.js
/*
 * Copyright (c) 2022 Maximilian Bauer
 * Distributed under the GNU GPL v3. For full terms see the file "COPYING".
 */
// @ts-check

import { Settings } from "./Settings.js";

export class HardwareSettings extends Settings {
   /** @type {number} */
   moistureSensorPin;
   /** @type {number} */
   dht22EchoPin;
   /** @type {number} */
   dht22TriggerPin;
   /** @type {number} */
   irrigatorSwitchPin;
   /** @type {number} */
   sensorSwitchPin;
   /** @type {number} */
   buzzerPin;
   /** @type {number} */
   hcsr04EchoPin;
   /** @type {number} */
   hcsr04TriggerPin;
   /** @type {number} */
   hcsr04MinimumValueCm;
   /** @type {number} */
   hcsr04MaximumValueCm;   
   /** @type {string} */
   password; 

   constructor(moistureSensorPin, dht22EchoPin, dht22TriggerPin, irrigatorSwitchPin, sensorSwitchPin, 
      buzzerPin, hcsr04EchoPin, hcsr04TriggerPin, hcsr04MinimumValueCm, hcsr04MaximumValueCm, password) {
      super();
      this.moistureSensorPin = moistureSensorPin;
      this.dht22EchoPin = dht22EchoPin;
      this.dht22TriggerPin = dht22TriggerPin;
      this.irrigatorSwitchPin = irrigatorSwitchPin;
      this.sensorSwitchPin = sensorSwitchPin;
      this.buzzerPin = buzzerPin;
      this.hcsr04EchoPin = hcsr04EchoPin;
      this.hcsr04TriggerPin = hcsr04TriggerPin;
      this.hcsr04MinimumValueCm = hcsr04MinimumValueCm;
      this.hcsr04MaximumValueCm = hcsr04MaximumValueCm;
      this.password = password;
   }

   clone() {
      return new HardwareSettings(this.moistureSensorPin, this.dht22EchoPin,
         this.dht22TriggerPin, this.irrigatorSwitchPin, this.sensorSwitchPin, 
         this.buzzerPin, this.hcsr04EchoPin, this.hcsr04TriggerPin,
         this.hcsr04MinimumValueCm, this.hcsr04MaximumValueCm, this.password);
   }
}